lib/ex_unit/test/ex_unit/capture_io_test.exs

```diff
@@ -28,6 +28,22 @@ defmodule ExUnit.CaptureIOTest do
   import ExUnit.CaptureIO
   doctest ExUnit.CaptureIO, import: true
 
+  test "no leakage on failures" do
+    processes_count = length(Process.list())
+    group_leader = Process.group_leader()
+
+    for _ <- 1..10 do
+      assert_raise ArgumentError, fn ->
+        capture_io(fn ->
+          raise ArgumentError
```

Comment:
Right, I also was unsure about `Process.list()`. I thought about processes check in the range, but your solution is way much better. Thank you.